sympy/polys/rootoftools.py

```diff
@@ -772,15 +775,94 @@ def _get_roots(cls, method, poly, radicals):
         free_names = {str(i) for i in poly.free_symbols}
         for x in chain((symbols('x'),), numbered_symbols('x')):
             if x.name not in free_names:
+                # xreplace makes alg field domain back into EX
+                # XXX: so make sure it stays same?
                 poly = poly.xreplace({d: x})
+                if dom.is_AlgebraicField:
+                    poly = Poly(poly, domain=dom)
                 break
+
+        if dom.is_QQ or dom.is_ZZ:
+            return cls._get_roots_qq(method, poly, radicals)
+            # XXX: any better way to check if real?
+            # without this check, some doctests fail (with complex terms)
+        if dom.is_AlgebraicField and all(c.is_real for c in poly.coeffs()):
+            return cls._get_roots_alg(method, poly, radicals)
```

Comment:
Why do we need to check `is_real` here?

Once we lift the polynomial the coefficients will be real.

Reply:
You're right, it's unnecessary.